bundle/definition/schema.go

```diff
@@ -0,0 +1,126 @@
+	Description          string                 `json:"description,omitempty" mapstructure:"description,omitempty"`
+	Else                 *Schema                `json:"else,omitempty" mapstructure:"else,omitempty"`
+	Enum                 []interface{}          `json:"enum,omitempty" mapstructure:"enum,omitempty"`
+	Examples             []interface{}          `json:"examples,omitempty" mapstructure:"examples,omitempty"`
+	ExclusiveMaximum     *float64               `json:"exclusiveMaximum,omitempty" mapstructure:"exclusiveMaximum,omitempty"`
+	ExclusiveMinimum     *float64               `json:"exclusiveMinimum,omitempty" mapstructure:"exclusiveMinimum,omitempty"`
+	Format               string                 `json:"format,omitempty" mapstructure:"format,omitempty"`
+	If                   *Schema                `json:"if,omitempty" mapstructure:"if,omitempty"`
+	//Items can be a Schema or an Array of Schema :(
+	Items         interface{} `json:"items,omitempty" mapstructure:"items,omitempty"`
+	Maximum       *float64    `json:"maximum,omitempty" mapstructure:"maximum,omitempty"`
+	MaxLength     *float64    `json:"maxLength,omitempty" mapstructure:"maxLength,omitempty"`
+	MinItems      *float64    `json:"minItems,omitempty" mapstructure:"minItems,omitempty"`
+	MinLength     *float64    `json:"minLength,omitempty" mapstructure:"minLength,omitempty"`
+	MinProperties *float64    `json:"minProperties,omitempty" mapstructure:"minProperties,omitempty"`
+	Minimum       *float64    `json:"minimum,omitempty" mapstructure:"minimum,omitempty"`
+	MultipleOf    *float64    `json:"multipleOf,omitempty" mapstructure:"multipleOf,omitempty"`
+	Not           *Schema     `json:"not,omitempty" mapstructure:"not,omitempty"`
+	OneOf         *Schema     `json:"oneOf,omitempty" mapstructure:"oneOf,omitempty"`
+
+	PatternProperties map[string]*Schema `json:"patternProperties,omitempty" mapstructure:"patternProperties,omitempty"`
+
+	Properties    map[string]*Schema `json:"properties,omitempty" mapstructure:"properties,omitempty"`
+	PropertyNames *Schema            `json:"propertyNames,omitempty" mapstructure:"propertyNames,omitempty"`
+	ReadOnly      *bool              `json:"readOnly,omitempty" mapstructure:"readOnly,omitempty"`
+	Required      []string           `json:"required,omitempty" mapstructure:"required,omitempty"`
+	Then          *Schema            `json:"then,omitempty" mapstructure:"then,omitempty"`
+	Title         string             `json:"title,omitempty" mapstructure:"title,omitempty"`
+	Type          interface{}        `json:"type,omitempty" mapstructure:"type,omitempty"`
+	UniqueItems   *bool              `json:"uniqueItems,omitempty" mapstructure:"uniqueItems,omitempty"`
+	WriteOnly     *bool              `json:"writeOnly,omitempty" mapstructure:"writeOnly,omitempty"`
+}
+
+// SingleType returns true if a schema has a single type declaration
+func (s *Schema) SingleType() bool {
```

Comment:
Is this function needs to be exported? It feels like it will only be used by `GetType` and `GetTypes` and not by end users who would mostly use `GetType` (but maybe I'm wrong).